nselib/http.lua

```diff
@@ -1571,7 +1571,11 @@ function parse_redirect(host, port, path, response)
   if ( not(u.host) ) then
     -- we're dealing with a relative url
     u.host = stdnse.get_hostname(host)
-    u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
+    if ( u.path:sub(1,1) == "/") then
+      u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
+    else
+      u.path = ((path:sub(1,1) == "/" and "" ) or "/" ) .. path .. ((path:sub(#path,#path) == "/" and "" ) or "/" ) .. u.path
```

Comment:
See my previous comment for this. The else statement will incorrectly append the path passed but this value can come from regular http.get() calls which will form an invalid path.